HVSample/src/main/java/com/microsoft/hsg/android/hvsample/RecordPickerArrayAdapter.java

```diff
@@ -18,45 +18,45 @@
 
 public class RecordPickerArrayAdapter extends BaseAdapter {
 	
-    private List<Record> records;
-    private HashMap<String, String> recordImages;
-    private LayoutInflater layoutInflater;
-    
-    private PersonalImageLoader imageLoader;
-
-    public RecordPickerArrayAdapter(Activity context, 
-    		List<Record> records,
-    		HealthVaultClient hvClient) 
-    {
-        this.records = records;
-        layoutInflater = LayoutInflater.from(context);
-        imageLoader = new PersonalImageLoader(context, hvClient);
-    }
-    
-    @Override
-    public int getCount() {
-        return records == null ? 0 : records.size();
-    }
-
-    @Override
-    public Record getItem(int position) {
-        return records.get(position);
-    }
-
-    @Override
-    public long getItemId(int position) {
-        return position;
-    }
-
-    @Override
-    public View getView(int position, View view, ViewGroup parent) {
-        View rowView= layoutInflater.inflate(R.layout.record_picker_item, null, true);
-        TextView txtTitle = (TextView) rowView.findViewById(R.id.txtrecordName);
-        ImageView imageView = (ImageView) rowView.findViewById(R.id.recordIcon);
-        txtTitle.setText(records.get(position).getName());
-        
-        imageLoader.load(records.get(position).getId(), imageView, R.drawable.ic_launcher);
-        
-        return rowView;
-    }
+	private List<Record> records;
+	private HashMap<String, String> recordImages;
+	private LayoutInflater layoutInflater;
+
+	private PersonalImageLoader imageLoader;
+
+	public RecordPickerArrayAdapter(Activity context,
+			List<Record> records,
+			HealthVaultClient hvClient)
+	{
+		this.records = records;
+		layoutInflater = LayoutInflater.from(context);
+		imageLoader = new PersonalImageLoader(context, hvClient);
+	}
+
+	@Override
+	public int getCount() {
+		return records == null ? 0 : records.size();
+	}
+
+	@Override
+	public Record getItem(int position) {
+		return records.get(position);
+	}
+
+	@Override
+	public long getItemId(int position) {
+		return position;
+	}
+
+	@Override
+	public View getView(int position, View view, ViewGroup parent) {
+		View rowView= layoutInflater.inflate(R.layout.record_picker_item, null, true);
```

Comment:
We need to use the view holder patter for any adapter which binds views like this. I remember you implementing it in another review. Inflating views is expensive and doing so in a list view will cause huge performance issues when scrolling. Anytime we create an adapter like this we should be using view holders. For listviews I would recommend using a recycleview instead. Recyclerview is more robust and helps force the viewholder pattern.